app/src/ui/app.tsx

```diff
@@ -290,11 +290,31 @@ export class App extends React.Component<IAppProps, IAppState> {
         return this.props.dispatcher.installCLI()
       case 'open-external-editor':
         return this.openCurrentRepositoryInExternalEditor()
+      case 'select-all':
+        return this.selectAll()
     }
 
     return assertNever(name, `Unknown menu event name: ${name}`)
   }
 
+  /**
+   * Handler for the 'select-all' menu event, dispatches
+   * a custom DOM event originating from the element which
+   * currently has keyboard focus. Components have a chance
+   * to intercept this event and implement their own 'select
+   * all' logic.
+   */
+  private selectAll() {
+    const event = new CustomEvent('select-all', {
+      bubbles: true,
+      cancelable: true,
+    })
+
+    if (document.activeElement.dispatchEvent(event)) {
```

Comment:
Can we be explicit about what version of false we're checking against.

Reply:
According to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/dispatchEvent) this seems to only be a `boolean`. Not sure what else we can do here...

Reply:
@niik, you can ignore this. shiftkey and I discussed earlier.